docs/examples/advanced_optimizations.ipynb

```diff
@@ -0,0 +1,330 @@
+    "\n",
+    "To show this in action, let's first initialize NCCL with a trivial process group:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "id": "fca06ec3",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Configure parallel groups\n",
+    "import os\n",
+    "import torch\n",
+    "world_group = torch.distributed.init_process_group(\n",
+    "    \"nccl\",\n",
+    "    init_method=\"file:///tmp/rdzv\",\n",
+    "    world_size=1,\n",
+    "    rank=0,\n",
+    ")\n",
+    "data_parallel_group = torch.distributed.new_group(ranks=[0], backend=\"nccl\")\n",
+    "tensor_parallel_group = torch.distributed.new_group(ranks=[0], backend=\"nccl\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1f2b80d0",
+   "metadata": {},
+   "source": [
+    "We only initialize with one GPU to keep this example simple. Please consult the documentation [torch.distributed](https://pytorch.org/docs/stable/distributed.html) for guidance on running with multiple GPUs. Note that we require that each distributed process corresponds to exactly one GPU, so we treat them interchangeably. In practice, there are multiple factors that can affect the optimal parallel layout: the system hardware, the network topology, usage of other parallelism schemes like pipeline parallelism. A rough rule-of-thumb is to interpret the GPUs as a 2D grid with dimensions of $\\text{num_nodes} \\times \\text{gpus_per_node}$. The rows are tensor-parallel groups and the columns are data-parallel groups.\n",
+    "\n",
+    "Enabling data parallelism with Transformer Engine is similar to enabling data parallelism with standard PyTorch models: simply wrap the modules with [torch.nn.parallel.DistributedDataParallel](https://pytorch.org/docs/stable/generated/torch.nn.parallel.DistributedDataParallel.html). FP8 training requires extra synchronization for the scaling factors, so the data-parallel process group must also be passed to the [fp8_autocast](../api/pytorch.rst#transformer_engine.pytorch.fp8_autocast) context manager. Transformer Engine modules also have native support for tensor and sequence parallelism. If the user provides a process group for tensor parallelism, the the modules will distribute the data and perform communication internally. If sequence parallelism is enabled, it will be applied for operations that are not amenable to tensor parallelism and it will use the tensor-parallel process group."
```

Comment:
```suggestion
    "Enabling data parallelism with Transformer Engine is similar to enabling data parallelism with standard PyTorch models: simply wrap the modules with [torch.nn.parallel.DistributedDataParallel](https://pytorch.org/docs/stable/generated/torch.nn.parallel.DistributedDataParallel.html). FP8 training requires extra synchronization for the scaling factors, so the data-parallel process group must also be passed to the [fp8_autocast](../api/pytorch.rst#transformer_engine.pytorch.fp8_autocast) context manager. Transformer Engine modules also have native support for tensor and sequence parallelism. If the user provides a process group for tensor parallelism, the modules will distribute the data and perform communication internally. If sequence parallelism is enabled, it will be applied for operations that are not amenable to tensor parallelism and it will use the tensor-parallel process group."
```